taroscript/send.go

```diff
@@ -0,0 +1,164 @@
+	)
+	for idx := range outputs {
+		vOut := outputs[idx]
+
+		// We assume the first output is the change output (or
+		// tombstone if there is no change in a non-interactive send).
+		if idx == 0 {
+			// The change output should always be at index 0.
+			if !vOut.IsChange {
+				return ErrInvalidChangeOutputLocation
+			}
+
+			// A zero-amount change output (tombstone) must spend to
+			// the un-spendable NUMS script key.
+			if vOut.Amount == 0 &&
+				!vOut.ScriptKey.PubKey.IsEqual(asset.NUMSPubKey) {
+
+				return commitment.ErrInvalidScriptKey
+			}
+		} else {
+			if vOut.IsChange {
+				return ErrInvalidChangeOutputLocation
+			}
+		}
+
+		residualAmount -= vOut.Amount
+
+		locator := vOut.SplitLocator(inputAsset.ID())
+		splitLocators[idx] = &locator
+	}
+
+	// We should now have exactly zero value left over after splitting.
+	if residualAmount != 0 {
+		return ErrInvalidSplitAmounts
+	}
+
+	// If we have an interactive full value send, we don't need a tomb stone
+	// at all.
+	inputIDCopy := input.PrevID
+	if interactiveFullValueSend(input, outputs) {
+		// We'll now create a new copy of the old asset, swapping out
+		// the script key. We blank out the tweaked key information as
+		// this is now an external asset.
+		outputs[1].Asset = inputAsset.Copy()
+		outputs[1].Asset.ScriptKey = outputs[1].ScriptKey
+
+		// Record the PrevID of the input asset in a Witness for the new
+		// asset. This Witness still needs a valid signature for the new
+		// asset to be valid.
+		//
+		// TODO(roasbeef): when we fix #121, then this should also be a
+		// ZeroPrevID
+		outputs[1].Asset.PrevWitnesses = []asset.Witness{
+			{
+				PrevID:          &inputIDCopy,
+				TxWitness:       nil,
+				SplitCommitment: nil,
+			},
+		}
+
+		// We are done, since we don't need to create a split
+		// commitment.
+		return nil
+	}
+
+	splitCommitment, err := commitment.NewSplitCommitment(
+		inputAsset, input.PrevID.OutPoint, splitLocators[0],
+		splitLocators[1:]...,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Assign each of the split assets to their respective outputs.
+	input.IsSplit = true
+	for idx := range outputs {
+		// The change output for a split asset send always gets the root
+		// asset committed, even if it's a zero value (tombstone) split
+		// output for the sender.
+		if outputs[idx].IsChange {
+			outputs[idx].Asset = splitCommitment.RootAsset.Copy()
+			continue
+		}
+
+		locator := splitLocators[idx]
+		splitAsset, ok := splitCommitment.SplitAssets[*locator]
+		if !ok {
+			return fmt.Errorf("invalid split, asset for locator "+
+				"%v not found", locator)
+		}
+
+		outputs[idx].Asset = &splitAsset.Asset
+		outputs[idx].Asset.ScriptKey = outputs[idx].ScriptKey
+	}
+
+	return nil
+}
+
+// interactiveFullValueSend returns true if the given outputs spend the input
+// fully and interactively.
+func interactiveFullValueSend(input *taropsbt.VInput,
+	outputs []*taropsbt.VOutput) bool {
+
+	return len(outputs) == 2 &&
```

Comment:
Shouldn't this be `outputs == 1`? So it would be the same as a consolidation transaction.

Reply:
I fixed the whole logic around the number of outputs. If it's a full value interactive send, there now can be just one output.